client/flutter/lib/pages/symptom_checker/symptom_checker_model.dart

```diff
@@ -38,25 +37,33 @@ class SymptomCheckerModel with ChangeNotifier {
   /// the pages list. If the series is complete the isComplete flag will be set.
   /// In both cases the change notifier will fire to indicate the update.
   void answerQuestion(Set<String> answerIds) {
+    try {
+      isComplete = _answerQuestionImpl(answerIds);
+    } catch (e) {
+      // Do NOT log these errors to analytics.
+      isFatalError = true;
+    }
+    notifyListeners();
+  }
+
+  bool _answerQuestionImpl(Set<String> answerIds) {
     pages[pages.length - 1] = currentPage.withAnswers(answerIds);
 
-    // TODO: Toy implementation - always add the next page if it exists
-    // TODO: This should check the display conditions.
-    // Add the next question
-    if (_content.questions.length > pages.length) {
-      pages.add(SymptomCheckerPageModel(
-          question: _content.questions[pages.length],
-          questionCount: _content.questions.length,
-          questionIndex: pages.length));
-    }
-    // TODO: Toy implementation - assume all pages are shown
-    // TODO: This should check the display conditions.
-    // No more questions, series complete.
-    if (_content.questions.length == pages.length &&
-        currentPage.selectedAnswers.isNotEmpty) {
-      isComplete = true;
+    bool nextPageFound = false;
+    for (var i = pages.length;
+        !nextPageFound && i < _content.questions.length;
+        i++) {
+      nextPageFound = SymptomLogic().evaluateCondition(
```

Comment:
Nit, but do we want to store a reference to an instance SymptomLogic rather than re-creating every time? Definitely not a blocker on this PR though

Reply:
Good point, done.